text/0000-katello-as-a-plugin.md

```diff
@@ -0,0 +1,107 @@
+- Start Date: 2016-09-09
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal of this effort is to allow Katello to be treated from a community and technical perspective as any other plugin.
+
+# Motivation
+[motivation]: #motivation
+
+Katello was originally a stand alone Rails application that attempted to integrate with Foreman through a single-sign-on process, Rails engines that knew how to communicate with one another and switching menu structures. After a desire to centralize on a single community and bring integration tighter, Katello was moved to be a plugin but given the constraints on Katello for depoyment along with timeline the deployment of Foreman and Katello was made unique when compared to any other plugin. 
+
+# Detailed design
+[design]: #detailed-design
+
+Given this is a large area of work with many facets, the design will list out each issue with proposed design solutions and any drawbacks. This may be able to be split out to their own RFCs over time.
+
+## Certificates
+
+Foreman defaults to using Puppet certificates for the web UI. Katello requires a certificate authority to hand to Candlepin for cutting client certificates and entitlement certificates. The same certificate authority used to cut client certificates is needed to have signed the server certificates so that clients can talk back to the server and access HTTPs based repositories. Further, certificates are needed to configure communication with Pulp, Qpid, Capsules (Smart Proxies) and creating entities like trustores for Candlepin.
+
+Another aspect of certificates is the desire by users to be able to supply their own certificates for the web UI. Users should be able to change those fluidly without having to propagate new certificates across their entire client infrastructure.
+
+#### Design
+
+Introduce the concept of an infrastructure port that the application runs on and uses the self-signed certificates for. This port would exist in a known range (e.g. 9070 - 9079) so that all other backend services could be similarly configured within this range with well defined port values. For example, Pulp could be deployed on 9072 and Candlepin 9073. All Smart Proxies, Capsules and Clients would talk to port 9070 and leave the web UI to run on 443 as is today. This would allow users to switch out the web UI certificates simply and easily while requiring no updates to the clients. Some discussion of this can be seen [here](https://github.com/theforeman/puppet-foreman/pull/374).
+
+#### Drawbacks
+
+This could introduce potential performance issues with running the application on two ports or complex configuration to ensure proper redirects. This change would introduce a one-time infrastructure update for existing users to move servers and Smart Proxies to the new port and certificate setup. Clients would also need updating to ensure they are pointing and talking to the correct port with RHSM. 
+
```

Comment:
I'm guessing this could be mitigated by not enabling this by default? Also, clients would still communicate to the infrastructure ports, so in general its only the web ui which has two ports by default?